File: app/src/main/java/com/example/hellofriend/Models/Message.java
package com.example.hellofriend.Models;

public class Message {

    private String text;
    private String userId;
    private String userName;
    private long timestamp;

    public Message() {
        // Default constructor required for calls to DataSnapshot.getValue(Message.class)
    }

    public Message(String text, String userId, String userName, long timestamp) {
        this.text = text;
        this.userId = userId;
        this.userName = userName;
        this.timestamp = timestamp;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public long getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(long timestamp) {
        this.timestamp = timestamp;
    }
}

File: app/src/main/java/com/example/hellofriend/Models/Message.kt
package com.example.hellofriend.Models

class Message {
    var text: String? = null
    var userId: String? = null
    var userName: String? = null
    var timestamp: Long = 0
    var recipientId: String? = null // Add this field

    constructor()

    constructor(
        text: String?,
        userId: String?,
        userName: String?,
        timestamp: Long,
        recipientId: String?
    ) {
        this.text = text
        this.userId = userId
        this.userName = userName
        this.timestamp = timestamp
        this.recipientId = recipientId // Initialize recipientId
    }

}
